DigitalLearningSolutions.Data/Services/RegistrationService.cs

```diff
@@ -0,0 +1,69 @@
+﻿namespace DigitalLearningSolutions.Data.Services
+{
+    using System;
+    using DigitalLearningSolutions.Data.DataServices;
+    using DigitalLearningSolutions.Data.Models.Email;
+    using DigitalLearningSolutions.Data.Models.Register;
+    using MimeKit;
+
+    public interface IRegistrationService
+    {
+        string RegisterDelegate(DelegateRegistrationModel delegateRegistrationModel, string baseUrl);
+    }
+    public class RegistrationService: IRegistrationService
+    {
+        private readonly IRegistrationDataService registrationDataService;
+        private readonly IPasswordDataService passwordDataService;
+        private readonly IEmailService emailService;
+        private readonly ICentresDataService centresDataService;
+
+        public RegistrationService(IRegistrationDataService registrationDataService,
+            IPasswordDataService passwordDataService, IEmailService emailService, ICentresDataService centresDataService)
+        {
+            this.registrationDataService = registrationDataService;
+            this.passwordDataService = passwordDataService;
+            this.emailService = emailService;
+            this.centresDataService = centresDataService;
+        }
+
+        public string RegisterDelegate(DelegateRegistrationModel delegateRegistrationModel, string baseUrl)
+        {
+            var candidateNumber = registrationDataService.RegisterDelegate(delegateRegistrationModel);

```

Comment:
Are the following returns handled (from the stored procedure documentation)?

```
'-1' : unexpected error
'-2' : some parameters not supplied
'-3' : failed, AliasID not unique
'-4' : active candidate record for e-mail address already exists 
```

I guess -2 should never be the case, but I can see the rest potentially coming up.